// ngx-highlight-js/bundles/highlightjs.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core'], factory) :
	(factory((global.ngxHighlightJS = {}),global.ng.core));
}(this, (function (exports,core) { 'use strict';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var HighlightJsDirective = /** @class */ (function () {
    function HighlightJsDirective(el) {
        this.el = el;
    }
    /**
     * @return {?}
     */
    HighlightJsDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.codeEl = document.createElement('pre');
        if (this.lang) {
            this.codeEl.className = this.lang;
        }
        this.codeEl.innerHTML = '' + this.el.nativeElement.innerHTML.trim();
        this.parentEl = this.el.nativeElement.parentNode;
        this.parentEl.insertBefore(this.codeEl, this.el.nativeElement.nextSibling);
        hljs.configure(Object.assign({}, this.options));
        hljs.highlightBlock(this.codeEl);
    };
    /**
     * @return {?}
     */
    HighlightJsDirective.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        if (this.codeEl) {
            this.parentEl.removeChild(this.codeEl);
            this.codeEl = null;
        }
    };
    HighlightJsDirective.decorators = [
        { type: core.Directive, args: [{
                    selector: '[highlight-js]',
                    host: {
                        style: 'display:none;',
                    },
                },] },
    ];
    /** @nocollapse */
    HighlightJsDirective.ctorParameters = function () { return [
        { type: core.ElementRef, },
    ]; };
    HighlightJsDirective.propDecorators = {
        "options": [{ type: core.Input },],
        "lang": [{ type: core.Input },],
    };
    return HighlightJsDirective;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var HighlightJsModule = /** @class */ (function () {
    function HighlightJsModule() {
    }
    HighlightJsModule.decorators = [
        { type: core.NgModule, args: [{
                    declarations: [HighlightJsDirective],
                    exports: [HighlightJsDirective]
                },] },
    ];
    return HighlightJsModule;
}());

exports.HighlightJsDirective = HighlightJsDirective;
exports.HighlightJsModule = HighlightJsModule;

Object.defineProperty(exports, '__esModule', { value: true });

})));
